### DISCUSSION TITLE
eShopSupport EmbeddingGenerationService

### DISCUSSION THREAD

**Discussion by Peppe426** [dotnet/eShop#491] [General]
Steven Sanderson gave an excellent presentation [here](https://www.youtube.com/watch?v=TSNAvFJoP4M) about useful AI in the eShopSupport soulution. Could we have peak at the **LocalTextEmbeddingGenerationService**? It does not have to be perfect. ⭐

**Comment by EwoutdBoer** [comment]
From the Ndc presentation and the dotnet blazor day I tried to implement the recommendation Python service, but got a couple of things not quite working. So I would like to peek at the code as well. @SteveSandersonMS mentioned that the code was under review so hoping (a part) of the samples will be available soon

**Comment by SteveSandersonMS** [comment]
Sorry the approval process has taken a while. We're at the end of it now and I'm really hoping we can make it public today, or failing that, within a couple of days.

**Reply by EwoutdBoer** [reply]
@SteveSandersonMS that is really good news. Thanks for the very quick feedback. If helpful, I could help with documentation to help out others not making same mistakes and errors I did

**Reply by kannan-cidc** [reply]
@SteveSandersonMS - was it approved yesterday ? eager to check the eshopSupport repo

**Reply by SteveSandersonMS** [reply]
Got the notification this morning that it was approved. Hopefully we will make it public later today, when the right people wake up to flip the switch :)

**Reply by SteveSandersonMS** [reply]
It's public now! https://github.com/dotnet/eShopSupport

**Comment by SteveSandersonMS** [comment]
Also if you just want to see `LocalTextEmbeddingGenerationService` specifically, that's already available under the name `LocalEmbedder` at https://github.com/dotnet/smartcomponents/blob/main/src/SmartComponents.LocalEmbeddings/LocalEmbedder.cs, with docs at https://github.com/dotnet/smartcomponents/blob/main/docs/local-embeddings.md.

**Comment by SteveSandersonMS** [comment]
Closing as done (https://github.com/dotnet/eShopSupport is live)
